cmd/cli/root.go

```diff
@@ -45,43 +42,21 @@ func NewRootCmd() *cobra.Command {
 		"api":     configflags.ClientAPIFlags,
 		"logging": configflags.LogFlags,
 	}
+	cfg := config.New()
 	RootCmd := &cobra.Command{
 		Use:   os.Args[0],
 		Short: "Compute over data",
 		Long:  `Compute over data`,
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
-			repoDir, err := config.Get[string]("repo")
-			if err != nil {
-				util.Fatal(cmd, fmt.Errorf("failed to read --repo value: %w", err), 1)
-			}
-			if repoDir == "" {
-				// this error indicates `defaultRepo` was unable to find a default location and the user
-				// didn't provide on.
-				util.Fatal(cmd, fmt.Errorf("bacalhau repo not set, please use BACALHAU_DIR or --repo"), 1)
-			}
-			if _, err := setup.SetupBacalhauRepo(repoDir); err != nil {
-				util.Fatal(cmd, fmt.Errorf("failed to initialize bacalhau repo at '%s': %w", repoDir, err), 1)
-			}
-
-			if err := configflags.BindFlags(cmd, rootFlags); err != nil {
+			if err := configflags.BindFlags(cmd, cfg, rootFlags); err != nil {
 				util.Fatal(cmd, err, 1)
 			}
 
-			// If a CA certificate was provided, it must be a file that exists. If it does not
-			// exist we should not continue.
-			if caCert, err := config.Get[string](types.NodeClientAPIClientTLSCACert); err == nil && caCert != "" {
-				if _, err := os.Stat(caCert); os.IsNotExist(err) {
-					util.Fatal(cmd, fmt.Errorf("CA certificate file '%s' does not exist", caCert), 1)
-				}
-			}
```

Comment:
Review: I can't imagine a more out-of-place location to perform this validation :face_with_spiral_eyes: I have removed this and add 1468f6d0968057049f184d03ffca12ee5b7ff970 to prevent the possibility of a panic in the client code.